src/notifications/eventsMap.ts

```diff
@@ -0,0 +1,26 @@
+import EventEmitter from 'events';
+
+export class EventsMap {
+  private registered: Map<string, EventEmitter>;
+
+  constructor() {
+    this.registered = new Map();
+  }
+  store(eventName: string, handler: (...args: any[]) => Promise<void>) {
+    if (!this.registered.has(eventName)) {
+      this.registered.set(eventName, new EventEmitter());
+    }
+
+    this.registered.get(eventName)?.on(eventName, async (...args: any[]) => {
+      try {
+        await handler(...args);
+      } catch (error) {
+        console.error(`Error in handler for event ${eventName}`, error);
```

Comment:
Client has a logger - it should be use also for `notification/webhook` stuff.

Reply:
fixed